mutiny-core/src/federation.rs

```diff
@@ -794,6 +897,46 @@ pub(crate) fn mnemonic_from_xpriv(xpriv: ExtendedPrivKey) -> Result<Mnemonic, Mu
     Ok(mnemonic)
 }
 
+async fn process_operation_until_finished(
+    logger: Arc<MutinyLogger>,
+    lightning_meta: LightningOperationMeta,
+    hash: [u8; 32],
+    operation_id: OperationId,
+    lightning_module: &Arc<fedimint_client::ClientModuleInstance<'_, LightningClientModule>>,
+) -> Option<MutinyInvoice> {
+    match lightning_meta.variant {
+        LightningOperationMetaVariant::Pay(pay_meta) => {
+            let invoice = convert_from_fedimint_invoice(&pay_meta.invoice);
+            if invoice.payment_hash().into_32() == hash {
+                match lightning_module.subscribe_ln_pay(operation_id).await {
+                    Ok(o) => Some(
+                        process_outcome(o, process_pay_state_ln, invoice, false, None, logger)
+                            .await,
+                    ),
+                    Err(_) => Some(invoice.into()),
+                }
+            } else {
+                None
+            }
+        }
+        LightningOperationMetaVariant::Receive { invoice, .. } => {
+            let invoice = convert_from_fedimint_invoice(&invoice);
+            if invoice.payment_hash().into_32() == hash {
+                match lightning_module.subscribe_ln_receive(operation_id).await {
+                    Ok(o) => Some(
+                        process_outcome(o, process_receive_state, invoice, true, None, logger)
+                            .await,
+                    ),
+                    Err(_) => Some(invoice.into()),
+                }
```

Comment:
Why return `Some(invoice)` on `Err` here?

Reply:
So we just get the latest state of the invoice in general, wherever it ended up as.

Reply:
Can you add a comment then